tests/app/data_model/test_session_store.py

```diff
@@ -1,270 +1,258 @@
-        self.assertEqual(session_store.session_data.tx_id, self.session_data.tx_id)
-
-    def _save_session(self, session_id, user_id, data, legacy=False):
-        raw_data = json_dumps(vars(data))
-        protected_header = {"alg": "dir", "enc": "A256GCM", "kid": "1,1"}
 
-        if legacy:
-            plaintext = base64url_encode(raw_data)
-        else:
-            plaintext = raw_data
 
-        jwe_token = jwe.JWE(
-            plaintext=plaintext, protected=protected_header, recipient=self.key
-        )
-
-        session_model = EQSession(
-            eq_session_id=session_id,
-            user_id=user_id,
-            session_data=jwe_token.serialize(compact=True),
-            expires_at=self.expires_at,
-        )
-        current_app.eq["storage"].put(session_model)
+def test_save(app, app_session_store):
+    with app.test_request_context():
+        app_session_store.session_store.create(
+            eq_session_id="eq_session_id",
+            user_id="test",
+            session_data=app_session_store.session_data,
+            expires_at=app_session_store.expires_at,
+        ).save()
+        session_store = SessionStore("user_ik", "pepper", "eq_session_id")
+        assert session_store.session_data.tx_id == "tx_id"
+
+
+def test_delete(app, app_session_store):
+    with app.test_request_context():
+        app_session_store.session_store.create(
+            eq_session_id="eq_session_id",
+            user_id="test",
+            session_data=app_session_store.session_data,
+            expires_at=app_session_store.expires_at,
+        ).save()
+        assert "test" == app_session_store.session_store.user_id
+        app_session_store.session_store.delete()
+        assert app_session_store.session_store.user_id is None
+
+
+def test_add_data_to_session(app, app_session_store):
+    with app.test_request_context():
+        app_session_store.session_store.create(
+            eq_session_id="eq_session_id",
+            user_id="test",
+            session_data=app_session_store.session_data,
+            expires_at=app_session_store.expires_at,
+        ).save()
+        display_address = "68 Abingdon Road, Goathill"
+        app_session_store.session_store.session_data.display_address = display_address
+        app_session_store.session_store.save()
+
+        session_store = SessionStore("user_ik", "pepper", "eq_session_id")
+        assert session_store.session_data.display_address == display_address
+
+
+def test_should_not_delete_when_no_session(app, app_session_store):
+    with app.test_request_context("/status") as context:
+        # Call clear with a valid user_id but no session in database
+        app_session_store.session_store.delete()
+
+        # No database calls should have been made
+        assert context.app.eq["storage"].client.delete_call_count == 0
+
+
+def test_session_store_ignores_new_values_in_session_data(app, app_session_store):
+    session_data = SessionData(
```

Comment:
This looks like it can be extracted

Reply:
Worth having a generic session data fixture.